clients/vault/src/system.rs

```diff
@@ -437,7 +437,8 @@ impl VaultService {
 		let is_public_network = wallet.is_public_network();
 
 		// re-submit transactions in the cache
-		wallet.resubmit_transactions_from_cache().await;
+		// todo: handle timeouts
```

Comment:
@b-yap 
do you still plan to handle somehow timeouts?

Reply:
good question. I'm not sure if it should be done in this PR, because there's 2 ways for this, [as Stellar suggested](https://developers.stellar.org/api/errors/http-status-codes/horizon-specific/timeout):
> 1. Keep resubmitting the same transaction (with the same sequence number) and wait until it finally is added to a new ledger, or
> 2. Increase the fee in order to prioritize the transaction.

If we choose the first one, next question would be: how many times should we keep resubmitting? What to do if it's really down? rewrite to cache again and simply crash the vault? Or go to the 2nd option?

If we choose the 2nd one, that means updating the envelope itself. We should alert the user about this. 

Maybe I'm overthinking and I need someone else's input. :neckbeard: @ebma

Reply:
I agree that this should not be handled in this PR but in a separate one. I extended issue #315 to not also introduce indefinite retries but also to add this to all places where we submit Stellar transactions.